src/api_call.js

```diff
@@ -1,9 +1,40 @@
 //@format
 
-export async function getPokeData() {
-  const response = await fetch('https://pokeapi.co/api/v2/type/grass/');
-  const responseJson = await response.json();
-  return responseJson.pokemon.slice(0, 3).map(eachResult => {
+export async function generateSearchResults(pokeType) {
+  const typeSet = await getTypeSet(pokeType);
+  const filteredTypeSet = filterPokeIds(typeSet);
+  const pokeStatsPromises = filteredTypeSet.map(eachId => {
+    return getPokemonStats(eachId);
+  });
+  const resolvedPokeStats = await Promise.all(pokeStatsPromises);
+  return resolvedPokeStats.map(eachPoke => {
+    return filterCharacteristics(eachPoke);
+  });
+}
+
+async function getTypeSet(pokeType) {
+  const response = await fetch(
+    'https://pokeapi.co/api/v2/type/' + pokeType + '/',
```

Comment:
I would set this in global variables/constants – host + endpoint. That way, if you want to change the API at any point, you don't have to go looking for the references all around the code :)